### PR TITLE
feat: implement tutorial data fetching

### PR DESCRIPTION
**Pull request by zchsh**
## ✅ "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## 🔗 Relevant links

- [Preview link][/waypoint] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Implements tutorial & collection data fetching for `views/product-landing` content.

## 🤷 Why

To prepare to render `TutorialCard` and `CollectionCard` components in the view.

## 🛠️ How

- Expands our content data type, and stubbed "block" components, to account for `collection_cards`
- Adds a helper `get-inline-content-maps` that receives arbitrary `data`, traverses for tutorial and collection slugs, and returns `{ inlineCollections, inlineTutorials }` objects which can be used to map slugs to tutorial or collection data
- Uses `get-inline-content-maps` in `transform-raw-content-to-prop` to transformed authored tutorial & collection slugs into formatted tutorial & collection card data
    - Uses the existing `formatTutorialCard` & `formatCollectionCard` for formatting

## 🧪 Testing

- [ ] Visit [/waypoint][/waypoint], and ensure the sidecar table-of-contents works as expected
- [ ] Visit [/vault][/vault], and ensure the sidecar table-of-contents works as expected

## 💭 Anything else?

Nope!

[task]: https://app.asana.com/0/1202097197789424/1202226088644868/f
[/waypoint]: https://dev-portal-git-zsproduct-landing-tutorial-data-hashicorp.vercel.app/waypoint
[/vault]: https://dev-portal-git-zsproduct-landing-tutorial-data-hashicorp.vercel.app/vault

### CODE REVIEW

File: src/data/vault-landing.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
         "url": "/vault/tutorials/getting-started"
       },
       {
-        "text": "CLI Quick Start",
+        "text": "Developer Quick Start",
```
Comment (zchsh) [inline src/data/vault-landing.json:24]: Nitpick on a copy-pasta slip-up from an earlier PR.

File: src/data/vault-landing.json [hunk 2]
```diff
@@ -37,6 +37,14 @@
       "type": "tutorial_cards",
       "tutorialSlugs": ["vault/transform", "vault/static-secrets"]
     },
+    {
+      "type": "collection_cards",
```
Comment (zchsh) [inline src/data/vault-landing.json:41]: I've added `collection_cards` blocks to both `/vault` & `/waypoint` content, even though they're not present in designs. Intent here is to
1. validate that the component & related data fetching works
2. make it clear to authors that both tutorial and collection cards are available for use.

File: src/views/product-landing/components/product-landing-blocks/blocks/collection-cards/index.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { CollectionCardsProps } from './types'
+import s from './collection-cards.module.css'
+
+function CollectionCards({ collectionCards }: CollectionCardsProps) {
```
Comment (zchsh) [inline src/views/product-landing/components/product-landing-blocks/blocks/collection-cards/index.tsx:4]: Another placeholder for `CollectionCards`, similar to the placeholders in #390 .

File: src/views/product-landing/components/product-landing-blocks/blocks/tutorial-cards/index.tsx [hunk 4]
```diff
@@ -1,11 +1,11 @@
 import { TutorialCardsProps } from './types'
 import s from './tutorial-cards.module.css'
 
-function TutorialCards({ tutorialSlugs }: TutorialCardsProps) {
+function TutorialCards({ tutorialCards }: TutorialCardsProps) {
```
Comment (zchsh) [inline src/views/product-landing/components/product-landing-blocks/blocks/tutorial-cards/index.tsx:4]: Now accepts card data, not slugs.

File: src/views/product-landing/components/product-landing-blocks/index.tsx [hunk 5]
```diff
@@ -18,7 +23,14 @@ function ProductLandingBlocks({ blocks }: ProductLandingBlocksProps) {
             )
           case 'tutorial_cards':
             return (
-              <TutorialCards key={idx} tutorialSlugs={block.tutorialSlugs} />
+              <TutorialCards key={idx} tutorialCards={block.tutorialCards} />
+            )
+          case 'collection_cards':
```
Comment (zchsh) [inline src/views/product-landing/components/product-landing-blocks/index.tsx:28]: New block type for `collection_cards`.

File: src/views/product-landing/helpers/get-inline-content-maps.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import gatherUniqueValues from 'lib/gather-unique-values'
+import {
+  getInlineCollections,
+  getInlineTutorials,
+  InlineCollections,
+  InlineTutorials,
+} from 'views/product-tutorials-view/helpers/get-inline-content'
+
+export async function getInlineContentMaps(data: unknown): Promise<{
```
Comment (zchsh) [inline src/views/product-landing/helpers/get-inline-content-maps.ts:9]: This new helper accepts some arbitrary `data`, traverses it, and returns `{ inlineTutorials, inlineCollections }`. Used to transform raw content (slugs) into tutorial & collection card data server-side.

File: src/views/product-landing/helpers/transform-raw-content-to-prop.ts [hunk 7]
```diff
@@ -39,12 +41,31 @@ export async function transformRawContentToProp(
...
   const transformedBlocks: ProductLandingBlock[] = blocks.map(
     (block: ProductLandingContentBlock) => {
       const { type } = block
       switch (type) {
         case 'heading':
           return { ...block, headingSlug: makeHeadingSlug(block.heading) }
+        case 'tutorial_cards':
+          return {
+            type: block.type,
+            tutorialCards: block.tutorialSlugs.map((slug: string) => {
```
Comment (zchsh) [inline src/views/product-landing/helpers/transform-raw-content-to-prop.ts:54]: For raw `tutorial_cards` and `collection_cards` content blocks, we map the author-provided slugs to card data, using the `INLINE_CONTENT` we fetched, as well as the `formatTutorialCard` and `formatCollectionCard` helpers from the respective target components we'll be rendering in the UI.

File: src/views/product-landing/schema.ts [hunk 8]
```diff
@@ -44,6 +44,18 @@ const TutorialCardsContentBlockSchema = Joi.object({
...
 })
 
+/**
+ * CollectionCards
+ */
+interface CollectionCardsContentBlock {
+  type: 'collection_cards'
+  collectionSlugs: string[]
+}
+const CollectionCardsContentBlockSchema = Joi.object({
```
Comment (zchsh) [inline src/views/product-landing/schema.ts:54]: We add collection cards to the content schema, and to the corresponding content type.

**Comment by zchsh** [comment]
@kendallstrautman This PR implements data fetching for tutorials content in `views/product-landing` content.

I've also added in a new `collection_cards` block type - this isn't in design specs, but seems almost certain authors will want it, and was relatively quick to add in.

Note: this PR targets `zs.product-landing-sidecar` / #390 for now as it relies on work in that PR to set up `transform-raw-content-to-props`. I'm thinking it makes sense to finalize #390 first, but wanted to get this on your radar as well since it's very related.

Let me know what you think 🙇

File: src/views/product-landing/helpers/get-inline-content-maps.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
+import gatherUniqueValues from 'lib/gather-unique-values'
```
Comment (kendallstrautman) [inline src/views/product-landing/helpers/get-inline-content-maps.ts:1]: That `gatherUniqueValues` function coming in handy! 💯

**Review by kendallstrautman — APPROVED** [review]
👏

**Pull request merged**